scratchai/learners/metrics.py

```diff
@@ -45,15 +45,41 @@ def miou(pred, gt, nc, c2n=None):
   if isinstance(gt, torch.Tensor):
     gt = gt.detach().clone().cpu().numpy()
   
-  iou = {}
-  miou = 0
-  for cls in range(nc):
-    inter = np.logical_and(pred == cls,  gt == cls).sum()
-    union = np.logical_or(pred == cls, gt == cls).sum()
-    iou[cls] = inter / union if inter != 0 and union != 0 else 1.
-    miou += iou[cls]
+  with torch.no_grad():
+    iou = {}
+    miou = 0
+    for cls in range(nc):
+      inter = np.logical_and(pred == cls,  gt == cls).sum()
+      union = np.logical_or(pred == cls, gt == cls).sum()
+      iou[cls] = inter / union if inter != 0 and union != 0 else 1.
+      miou += iou[cls]
   
   # TODO print the iou in table format
   #print (tabulate([list(iou.keys()), list(iou.values())]))
 
   return miou/nc
+
+
+def accuracy(pred, gt, k=1):
```

Comment:
Accuracy calculated here: https://github.com/iArunava/scratchai/blob/e2fdc8f47232a18231975e176749c9ab4e9dcfe6/scratchai/learners/clflearner.py#L64-L67

Put the code in the above lines in this function, later this function can be made more general to return top-1 and top-5 accuracy.